app/helpers/admin/products_helper.rb

```diff
@@ -22,7 +22,8 @@ def unit_value_with_description(variant)
       precised_unit_value = number_with_precision(
         scaled_unit_value,
         precision: nil,
-        strip_insignificant_zeros: true
+        strip_insignificant_zeros: true,
+        significant: false,
```

Comment:
According to the documentation, `significant` is false by default. So this wouldn't change anything. Do you have a test for this?

https://api.rubyonrails.org/classes/ActiveSupport/NumberHelper.html#method-i-number_to_rounded

Reply:
This [error](https://github.com/openfoodfoundation/openfoodnetwork/issues/12739) keeps popping up on the Hungary instance when we don't explicitly set significant false (despite what the documentation says). 
I don't think we've got to the bottom of the reason yet. 
@rioug spent a little time trying to get to the bottom of it, I think he wasn't able to conclusively call it a Rails bug. So it's possibly something related to the OFN codebase somehow.

It does seem likely that this could happen again (anybody could use number_with_precision in the future, and not realise about this obscure issue). But I'm not sure how best to proceed. We could try monkey-patching, but that's not a great solution.. 

It would be better to understand why it's happening first. A couple of thoughts:
* replicate with a spec 
* look further into the rails source code to understand what's happening

Note that there's a few HU-specific variables that could be related. I'm guessing it's the system locale:
https://github.com/openfoodfoundation/ofn-install/blob/master/inventory/group_vars/hu.yml

I started digging into the source code, but it seems that `significant` always defaults to `false`, so I don't know what's happening or where the locale comes into play.
https://github.com/rails/rails/blob/main/activesupport/lib/active_support/number_helper/number_converter.rb